commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -218,3 +219,31 @@ def get_orders_for_customer(self, edx_lms_user_id: int, offset=0,
         orders = self.get_orders(customer, offset, limit)
 
         return orders, customer
+
+    def get_product_variant_by_course_run(self, cr_id: str) -> Optional[CTProductVariant]:
+        results = self.base_client.product_projections.search(False, filter=f"variants.sku:\"{cr_id}\"").results
+
+        if len(results) < 1:
+            return None
+
+        # Make 2D List of all variants from all results, and then flatten
+        all_variants = [listitem for sublist in
+                        list(
+                            map(
+                                lambda selection: [selection.master_variant, *selection.variants],
+                                results
+                            )
+                        )
+                        for listitem in sublist]
+
+        matching_variant_list = list(
+            filter(
+                lambda v: v.sku == cr_id,
+                all_variants
+            )
+        )
+
+        if len(matching_variant_list) < 1:
+            return None
+
+        return matching_variant_list[0]
```

Comment:
IMO, we can avoid this further filtering of data to determine whether the product exists or not.

In the case where the product by variant exists, it will return in the `results` object and that should be enough for us to determine the variant's existence

In the case where product by variant does not exist, it will return an empty list `[]`

Reply:
This was a temporary function to implement in the `GetActiveOrderManagementSystem` pipeline. @grmartin created the file `edx_utils.py` which consists of all the edx helper functions which I will use in the pipeline instead.

Reply:
The filtering makes it easier to access certain attributes of the product variant object for testing purposes. For example, if not for this filtering, I would not be able to access the sku of the product variant object which is is used in the unit tests.

Reply:
Shafqat, this is also a FT Search... so its using inexact matching (think like soundex) so we need to search the return to ensure an exact match is found... hence the filter.